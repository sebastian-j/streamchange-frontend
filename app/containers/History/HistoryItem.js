import React from 'react';
import PropTypes from 'prop-types';
import RelativeDate from '../../components/RelativeDate';

const HistoryItem = props => (
  <tr className="md-tr">
    <td className="md-td md-td-image">
      <a
        href={`https://www.youtube.com/channel/${props.channelId}`}
        target="_blank"
      >
        <img alt="Logo" src={props.imageUrl} height="45px" />
      </a>
    </td>
    <td className="md-td md-td-text-left">
      <span>{props.displayName}</span>
    </td>
    <td className="md-td md-td-text">
      <span>{props.prize}</span>
    </td>
    <td className="md-td md-td-text">
      <span>{props.message}</span>
    </td>
    <td className="md-td md-td-text">
      <RelativeDate ISO8601Date={props.createdAt} />
    </td>
  </tr>
);

HistoryItem.propTypes = {
  channelId: PropTypes.string.isRequired,
  imageUrl: PropTypes.string,
  displayName: PropTypes.string,
  prize: PropTypes.string,
  message: PropTypes.number,
  createdAt: PropTypes.string,
};
export default HistoryItem;
