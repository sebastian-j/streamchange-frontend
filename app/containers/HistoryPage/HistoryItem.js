import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import RelativeDate from '../../components/RelativeDate';

const Row = styled.tr`
  height: 50px;
`;

const Cell = styled.td`
  border-bottom: 1px solid #5e5e5e;
  border-spacing: 0;
  border-collapse: collapse;
  color: ${(props) => props.theme.staticTextColor};
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  ${({ image }) =>
    image &&
    `
    max-width: 100px;
  `}
  ${({ textLeft }) =>
    textLeft &&
    `
    font-size: 1rem;
    padding: 14px 30px 14px 16px;
    text-align: left;
  `}
  ${({ text }) =>
    text &&
    `
    font-size: 0.875rem;
    padding: 14px 30px 14px 16px;
    text-align: right;
  `}
`;

const HistoryItem = (props) => (
  <Row>
    <Cell image>
      <a
        href={`https://www.youtube.com/channel/${props.channelId}`}
        target="_blank"
      >
        <img alt="Logo" src={props.imageUrl} height="45px" />
      </a>
    </Cell>
    <Cell textLeft>
      <span>{props.displayName}</span>
    </Cell>
    <Cell text>
      <span>{props.prize}</span>
    </Cell>
    <Cell text>
      <span>{props.message}</span>
    </Cell>
    <Cell text>
      <RelativeDate ISO8601Date={props.createdAt} />
    </Cell>
  </Row>
);

HistoryItem.propTypes = {
  channelId: PropTypes.string.isRequired,
  imageUrl: PropTypes.string,
  displayName: PropTypes.string,
  prize: PropTypes.string,
  message: PropTypes.string,
  createdAt: PropTypes.string,
};
export default HistoryItem;
