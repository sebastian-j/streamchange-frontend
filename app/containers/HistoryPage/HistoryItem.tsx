import React from 'react';
import PropTypes from 'prop-types';

import { Cell } from './components/Cell';
import { Row } from './components/Row';
import RelativeDate from '../../components/RelativeDate';

const HistoryItem = (props) => (
  <Row>
    <Cell className="image">
      <a
        href={`https://www.youtube.com/channel/${props.channelId}`}
        target="_blank"
      >
        <img alt="Logo" src={props.imageUrl} height="45px" />
      </a>
    </Cell>
    <Cell className="textLeft">
      <span>{props.displayName}</span>
    </Cell>
    <Cell className="text">
      <span>{props.prize}</span>
    </Cell>
    <Cell className="text">
      <span>{props.message}</span>
    </Cell>
    <Cell className="text">
      <RelativeDate ISO8601Date={props.createdAt} />
    </Cell>
  </Row>
);

HistoryItem.propTypes = {
  channelId: PropTypes.string.isRequired,
  imageUrl: PropTypes.string,
  displayName: PropTypes.string,
  prize: PropTypes.string,
  message: PropTypes.string,
  createdAt: PropTypes.string,
};
export default HistoryItem;
