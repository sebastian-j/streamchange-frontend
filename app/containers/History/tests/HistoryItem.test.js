import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import HistoryItem from '../HistoryItem';

const renderer = new ShallowRenderer();

describe('<HistoryItem />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(
      <HistoryItem
        channelId="id"
        displayName="abc"
        message="abc"
        prize="prize"
        imageUrl="url"
        createdAt="2019-12-24T07:27:56.273Z"
      />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
