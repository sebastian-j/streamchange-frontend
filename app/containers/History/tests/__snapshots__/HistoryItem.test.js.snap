// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryItem /> should render and match the snapshot 1`] = `
<ForwardRef>
  <ForwardRef
    image={true}
  >
    <a
      href="https://www.youtube.com/channel/id"
      target="_blank"
    >
      <img
        alt="Logo"
        height="45px"
        src="url"
      />
    </a>
  </ForwardRef>
  <ForwardRef
    textLeft={true}
  >
    <span>
      abc
    </span>
  </ForwardRef>
  <ForwardRef
    text={true}
  >
    <span>
      prize
    </span>
  </ForwardRef>
  <ForwardRef
    text={true}
  >
    <span>
      abc
    </span>
  </ForwardRef>
  <ForwardRef
    text={true}
  >
    <RelativeDate
      ISO8601Date="2019-12-24T07:27:56.273Z"
      className="relativeDate"
    />
  </ForwardRef>
</ForwardRef>
`;
